vcd/resource_vcd_edgegateway.go

```diff
@@ -17,12 +17,12 @@ func resourceVcdEdgeGateway() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"name": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```

Comment:
It would help to have a `Description` here explaining why the name can be optional, which is quite different from many of the other resources.

Reply:
it was a typo. Fixed